src/lib/eventManager.js

```diff
@@ -23,18 +33,50 @@ function handleEvent(event) {
   // 2. 각 요소에 대해 해당 이벤트 타입의 핸들러가 있는지 확인
   // 3. 핸들러가 있다면 실행
   // 이를 통해 하위 요소에서 발생한 이벤트를 상위에서 효율적으로 처리할 수 있습니다.
+
+  let target = event.target;
+
+  while (target && target !== rootElement) {
+    const handlers = eventMap.get(event.type);
+    if (handlers) {
+      const handler = handlers.get(target);
+      if (handler) handler(event);
+    }
+    target = target.parentNode;
+  }
 }
 
 // TODO: addEvent 함수 구현
 export function addEvent(element, eventType, handler) {
   // 1. eventMap에 이벤트 타입과 요소, 핸들러 정보 저장
   // 2. 필요한 경우 루트 요소에 새 이벤트 리스너 추가
   // 이 함수를 통해 개별 요소에 직접 이벤트를 붙이지 않고도 이벤트 처리 가능
+
+  if (!eventMap.has(eventType)) {
+    eventMap.set(eventType, new Map());
+  }
+  const handlers = eventMap.get(eventType);
+  handlers.set(element, handler);
+
+  if (rootElement) {
+    rootElement.addEventListener(eventType, handleEvent, true);
+  }
 }
 
 // TODO: removeEvent 함수 구현
 export function removeEvent(element, eventType, handler) {
   // 1. eventMap에서 해당 요소와 이벤트 타입에 대한 핸들러 제거
   // 2. 해당 이벤트 타입의 모든 핸들러가 제거되면 루트 요소의 리스너도 제거
   // 이를 통해 더 이상 필요 없는 이벤트 핸들러를 정리하고 메모리 누수 방지
+
+  const handlers = eventMap.get(eventType);
+  if (handlers) {
+    handlers.delete(element);
+    if (handlers.size === 0) {
+      eventMap.delete(eventType);
+      if (rootElement) {
+        rootElement.removeEventListener(eventType, handleEvent);
+      }
```

Comment:
```suggestion
        rootElement?.removeEventListener(eventType, handleEvent);
```

이렇게 `?`를 사용하면 if문이 계속 중첩되는 것보다 훨씬 깔끔하고 가독성도 좋아지더라구요! 저도 자주 쓰는 방식이에요 ㅎㅎ

Reply:
오호 그렇네요..! 저도 자주 사용해야겠어요:)